File: 70. Climbing Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int dp1 = 1, dp2 = 1;
        int i = 1;
        while (i < n) {
            int tmp = dp2;
            dp2 += dp1;
            dp1 = tmp;
            i++;
        }
        return dp2;
    }
};

File: java/70. Climbing Stairs.java
// O(n)
class Solution {
    public int climbStairs(int n) {
        int l1 = 1, l2 = 1;
        for (int i = 2; i <= n; ++i) {
            int temp = l2 + l1;
            l1 = l2;
            l2 = temp;
        }
        return l2;
    }
}
